src/negotiation/maker_negotiator.ts

```diff
@@ -120,28 +151,40 @@ export class MakerHttpApi {
     });
 
     app.get("/orders/:tradingPair/:orderId/executionParams", async (_, res) => {
-      res.send(this.maker.getExecutionParams());
+      res.send(this.getExecutionParams());
     });
 
     app.post("/orders/:tradingPair/:orderId/take", async (req, res) => {
-      const order = this.maker.getOrderById(req.params.orderId);
+      const order = this.getOrderById(req.params.orderId);
       const body = req.body;
 
       if (!order || req.params.tradingPair !== order.tradingPair) {
         res.status(404).send("Order not found");
       } else if (!body || !body.swapId) {
         res.status(400).send("swapId missing from payload");
       } else {
-        res.send(this.maker.takeOrder(body.swapId, order));
+        res.send(this.takeOrder(body.swapId, order));
       }
     });
 
-    app.listen(port, () =>
+    this.server = app.listen(port, () =>
       console.log(`Maker's Negotiation Service is listening on port ${port}.`)
     );
   }
 
-  public addOrder(order: Order) {
-    this.maker.addOrder(order);
+  public getUrl(): undefined | string {
+    if (this.server) {
+      const addr = this.server.address();
```

Comment:
Do you know what address this is returning? 
I could imagine that this would return the address it is listening on, e.g .`0.0.0.0` on default. 

An alternative approach would be to explicitly define the listening hostname as described here: 
https://nodejs.org/api/synopsis.html

e.g. 
```
const http = require('http');

const hostname = '127.0.0.1';
const port = 3000;

const server = http.createServer((req, res) => {
  res.statusCode = 200;
  res.setHeader('Content-Type', 'text/plain');
  res.end('Hello, World!\n');
});

server.listen(port, hostname, () => {
  console.log(`Server running at http://${hostname}:${port}/`);
});
```

Reply:
It returns IPV6 localhost at the moment.
I think the ultimate approach would be to either:
- Use libp2p
- Allow user to configure what is the external ip of the maker

Reply:
Interestingly enought, `app.listen(3218, "127.0.0.1)` does not work, making my life more complicated.

Reply:
I made something, have a look please.

Reply:
> It returns IPV6 localhost at the moment.
> I think the ultimate approach would be to either:
> 
> * Use libp2p
> * Allow user to configure what is the external ip of the maker
> 
> Even if we allow the user to pass an ip to the listen function (and I can do that easily) this ultimately may not help behind a nat etc.

libp2p probably will make it worse first as people are not used to the multi-address format. 

Feel free to keep it as it is if it works in a useful way. 
If we only optimize for these two lines of code then we probably don't need this function at all:

https://github.com/comit-network/create-comit-app/blob/35f956e0ffd0e47be13e58cf85e924988ec6175c/create/new_project/examples/btc_eth/src/maker.ts#L85-L88

We should keep in mind that the sdk should be useful outside of the examples. 
I have the feelings we trim the sdk to make life easy for the examples but forget about real world usability. 

What do you think?

p.s. sorry, I think we should have discussed this earlier.

Reply:
> > It returns IPV6 localhost at the moment.
> > I think the ultimate approach would be to either:
> > 
> > * Use libp2p
> > * Allow user to configure what is the external ip of the maker
> > 
> > Even if we allow the user to pass an ip to the listen function (and I can do that easily) this ultimately may not help behind a nat etc.
> 
> libp2p probably will make it worse first as people are not used to the multi-address format.
> 
> Feel free to keep it as it is if it works in a useful way.
> If we only optimize for these two lines of code then we probably don't need this function at all:
> 
> [comit-network/create-comit-app:create/new_project/examples/btc_eth/src/maker.ts@`35f956e`#L85-L88](https://github.com/comit-network/create-comit-app/blob/35f956e0ffd0e47be13e58cf85e924988ec6175c/create/new_project/examples/btc_eth/src/maker.ts#L85-L88)
> 
> We should keep in mind that the sdk should be useful outside of the examples.
> I have the feelings we trim the sdk to make life easy for the examples but forget about real world usability.
> 
> What do you think?
> 
> p.s. sorry, I think we should have discussed this earlier.

Did you look at my last commit? https://github.com/comit-network/comit-js-sdk/pull/100/commits/d0d12011831d7d27ce3396a1c73a66c82e0d74cb

Reply:
Yes, saw it. Looks good.

What do you think about my point?

Reply:
I am not sure I understand your point.
I feel like `getUrl()` has some usefulness outside the sdk.

Reply:
Let's not over discuss this topic, it is such a small change.

It's good if you see the usefulness outside of the examples. That's what I wanted to point out. 

So let's keep it as you propose.